examples/wallet/app/models/index.js

```diff
@@ -9,3 +9,22 @@ export type PrivateKey = string;
 export type Identifier = string;
 export type Delegation = Array<{ poolId: string, amount: number }>;
 export type PoolId = string;
+export type TransactionHash = string;
+export type CertificateType =
+  | 'STAKE_DELEGATION'
+  | 'POOL_REGISTRATION'
+  | 'OWNER_STAKE_DELEGATION';
+export type Certificate = {
+  type: CertificateType,
+  pool: PoolId
+};
+
+export type Transaction = {
+  id: TransactionHash,
+  certificate?: Certificate,
+  blockHeight?: number, // it will not be present until the transaction is included in a block
+  inputs: Array<TransactionInput>,
+  outputs?: Array<TransactionOutput>
```

Comment:
this means outputs is always non empty when present?

Reply:
the `Array<type>` does not require it being non-empty, so this means it might not be present, and it might be present but be empty.
I'll see if the node returns an empty array when there are no outputs, and make it non-optional if it does.